### PR TITLE
[Resource] Add user tags to workspace resource

### PR DESCRIPTION
**Pull request by ismirlia**
Adds user tags argument
Deprecates workspace details attribute.

```
--- PASS: TestAccIBMPIWorkspaceBasic (134.98s)
PASS
--- PASS: TestAccIBMPIWorkspaceUserTags (133.32s)
PASS
```

### CODE REVIEW

**Comment by ismirlia** [comment]
@yussufsh, @hkantare Can you please review this PR which part of power Q3

File: ibm/service/power/resource_ibm_pi_workspace.go [hunk 1]
```diff
@@ -199,3 +235,17 @@ func isIBMPIResourceDeleteRefreshFunc(client *instance.IBMPIWorkspacesClient, id
 		}
 	}
 }
+
+func resourceIBMPIWorkspaceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.HasChange(Arg_UserTags) {
+		crn := d.Get(Attr_CRN)
+		if crn != "" {
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_workspace.go:242]: I believe you also need crn != nil since you are reading from d.get

Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_workspace.go]: updated

File: ibm/service/power/resource_ibm_pi_workspace.go [hunk 2]
```diff
@@ -89,16 +104,28 @@ func resourceIBMPIWorkspaceCreate(ctx context.Context, d *schema.ResourceData, m
...
 		return diag.FromErr(err)
 	}
 
+	// Add user tags for newly created workspace
+	if tags, ok := d.GetOk(Arg_UserTags); ok {
+		if len(flex.FlattenSet(tags.(*schema.Set))) > 0 {
+			oldList, newList := d.GetChange(Arg_UserTags)
+			err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, *controller.CRN, "", UserTagType)
+			if err != nil {
+				log.Printf("Error on creation of workspace (%s) user_tags: %s", *controller.CRN, err)
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_workspace.go:121]:
```suggestion
				log.Printf("Error on creation of workspace (%s) pi_user_tags: %s", *controller.CRN, err)
```

File: ibm/service/power/resource_ibm_pi_workspace.go [hunk 3]
```diff
@@ -137,11 +164,19 @@ func resourceIBMPIWorkspaceRead(ctx context.Context, d *schema.ResourceData, met
 		return diag.FromErr(err)
 	}
 	d.Set(Arg_Name, controller.Name)
+	tags, err := flex.GetGlobalTagsUsingCRN(meta, *controller.CRN, "", UserTagType)
+	if err != nil {
+		log.Printf("Error on get of workspace (%s) user_tags: %s", cloudInstanceID, err)
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_workspace.go:169]:
```suggestion
		log.Printf("Error on get of workspace (%s) pi_user_tags: %s", cloudInstanceID, err)
```

File: ibm/service/power/resource_ibm_pi_workspace.go [hunk 4]
```diff
@@ -199,3 +234,16 @@ func isIBMPIResourceDeleteRefreshFunc(client *instance.IBMPIWorkspacesClient, id
...
 	}
 }
+
+func resourceIBMPIWorkspaceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.HasChange(Arg_UserTags) {
+		if crn, ok := d.GetOk(Attr_CRN); ok {
+			oldList, newList := d.GetChange(Arg_UserTags)
+			err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, crn.(string), "", UserTagType)
+			if err != nil {
+				log.Printf("Error on update of workspace (%s) user_tags: %s", crn, err)
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_workspace.go:244]:
```suggestion
				log.Printf("Error on update of workspace (%s) pi_user_tags: %s", crn, err)
```

File: ibm/service/power/resource_ibm_pi_workspace_test.go [hunk 5]
```diff
@@ -34,14 +35,41 @@ func TestAccIBMPIWorkspaceBasic(t *testing.T) {
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccIBMPIWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPIWorkspaceUserTagConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPIWorkspaceExists("ibm_pi_workspace.powervs_service_instance"),
+					resource.TestCheckResourceAttrSet("ibm_pi_workspace.powervs_service_instance", "id"),
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_workspace_test.go:49]: Should there be a check for `pi_user_tags` to be set correctly?

File: ibm/service/power/resource_ibm_pi_workspace.go (hunk 2 above)
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_workspace.go:121]: updated error messages

File: ibm/service/power/resource_ibm_pi_workspace.go (hunk 3 above)
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_workspace.go:169]: updated error messages

**Review by Alexander-Kita — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Review by yussufsh — APPROVED** [review]

**Pull request merged**